webapp-client/src/component/CategoryList.js

```diff
@@ -1,30 +1,73 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import allowErrorMessage from '../errorify';
 
 export default class CategoryList extends React.Component {
-  constructor(props){
+  constructor(props) {
     super(props);
-    this.addAElement = this.addAElement.bind(this);
+    this.state = {
+      hasError: false,
+      categoryList: [],
+    };
+    this.addCategoryElement = this.addCategoryElement.bind(this);
+    allowErrorMessage(this);
   }
 
-  addAElement(category, index){
+  async componentDidMount() {
+    const header = { 'Content-Type': 'application/json' };
+    try {
+      const response = await fetch(
+        process.env.REACT_APP_API_URL +
+          '/category/getCategories/' +
+          this.props.userID,
+        header
+      );
+      if (response.status !== 200) {
+        throw new Error(response.statusMessage);
+      }
+      const categoryList = await response.json();
```

Comment:
Can you add a comment describing what this object looks like or referencing where it is defined or what type it is?